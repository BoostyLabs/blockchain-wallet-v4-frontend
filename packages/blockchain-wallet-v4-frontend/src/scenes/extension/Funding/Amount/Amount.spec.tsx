import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Amount } from '.'

describe('Amount', (): void => {
  const component = shallow(<Amount />)
  const tree = toJson(component)

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot()
  })

  it('corrent input value', () => {
    const amount = '12.3456'
    component.find('#amountInput').simulate('change', {
      target: {
        value: amount
      }
    })
    expect(component.find('#amountInput').prop('value')).toEqual(amount)
  })

  it('corrent exchanged value', () => {
    const amount = '10'
    const exchangeRate = 2400
    component.find('#amountInput').simulate('change', {
      target: {
        value: amount
      }
    })

    expect(Number(amount) / exchangeRate).toEqual(
      parseFloat(component.find('#exchangedAmount').text())
    )
  })
})
