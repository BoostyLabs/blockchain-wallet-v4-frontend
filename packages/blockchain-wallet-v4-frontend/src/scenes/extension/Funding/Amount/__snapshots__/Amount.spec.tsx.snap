// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount renders correctly 1`] = `
<Fragment>
  <Styled(Text)>
    Enter amount
  </Styled(Text)>
  <styled.div>
    <styled.input
      id="amountInput"
      onChange={[Function]}
      type="number"
      value="0"
    />
    <styled.span
      onClick={[Function]}
    >
      USD 
      <SvgIconChevronDown />
    </styled.span>
  </styled.div>
  <styled.div
    id="exchangedAmount"
  >
    0 ETH
  </styled.div>
  <styled.div
    key="Gas Fee"
    style={
      Object {
        "margin": "30px 0",
      }
    }
  >
    <Text
      altFont={false}
      capitalize={false}
      color="#fff"
      cursor="inherit"
      italic={false}
      lineHeight="24px"
      opacity={1}
      size="16px"
      uppercase={false}
      weight={400}
    >
      Gas Fee
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="#fff"
      cursor="inherit"
      italic={false}
      lineHeight="24px"
      opacity={1}
      size="16px"
      style={
        Object {
          "textAlign": "right",
        }
      }
      uppercase={false}
      weight={400}
    >
      5.3655 ETH
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="16px"
      uppercase={false}
      weight={400}
    />
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="20px"
      opacity={1}
      size="14px"
      style={
        Object {
          "textAlign": "right",
        }
      }
      uppercase={false}
      weight={400}
    >
      5.3655 USD
    </Text>
  </styled.div>
  <styled.div
    key="Total"
    style={
      Object {
        "margin": "30px 0",
      }
    }
  >
    <Text
      altFont={false}
      capitalize={false}
      color="#fff"
      cursor="inherit"
      italic={false}
      lineHeight="24px"
      opacity={1}
      size="16px"
      uppercase={false}
      weight={400}
    >
      Total
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="#fff"
      cursor="inherit"
      italic={false}
      lineHeight="24px"
      opacity={1}
      size="16px"
      style={
        Object {
          "textAlign": "right",
        }
      }
      uppercase={false}
      weight={400}
    >
      5.3655 ETH
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="16px"
      uppercase={false}
      weight={400}
    />
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="20px"
      opacity={1}
      size="14px"
      style={
        Object {
          "textAlign": "right",
        }
      }
      uppercase={false}
      weight={400}
    >
      5.3655 USD
    </Text>
  </styled.div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Styled(Link)
      style={
        Object {
          "background": "none",
          "color": "#65A5FF",
          "marginRight": "10px",
        }
      }
      to="/extension/funding/select-account"
    >
      Cancel
    </Styled(Link)>
    <Styled(Link)
      to="/extension/funding/success"
    >
      Confirm
    </Styled(Link)>
  </div>
</Fragment>
`;
